.github/workflows/listgen.yml

```diff
@@ -0,0 +1,115 @@
+name: Generate stdlib lists
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * 0"
+
+jobs:
+  pre-list-legacy:
+    strategy:
+      matrix:
+        python:
+          - "3.7"
+          - "3.8"
+          - "3.9"
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          # NOTE: matrix.python is intentionally not used here.
+          python-version: "3.x"
+      - name: support deps
+        run: make dev INSTALL_EXTRA=support
+      - name: build pre-list
+        env:
+          LISTGEN_PYTHON_VERSION: "${{ matrix.python }}"
+        run: |
+          ./env/bin/python ./support/fetch-sphinx.py "${LISTGEN_PYTHON_VERSION}" > pre-list.txt
+      - name: upload pre-list
+        uses: actions/upload-artifact@v3
+        with:
+          name: pre-list-${{ matrix.python }}
+          path: pre-list.txt
+
+  expand-list-legacy:
+    needs: pre-list-legacy
+    strategy:
+      matrix:
+        python:
+          - "3.7"
+          - "3.8"
+          - "3.9"
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏the pre-list was to fetch all existing inventory, and the `expand` approach is to augment them with the modules found in the runtime version?
If that's true, there are other modules/submodules that exist in other runtimes for non-linux Python distributions. Are those out of scope for now? If so, should we track that for later?

Reply:
> **question:** ‏the pre-list was to fetch all existing inventory, and the `expand` approach is to augment them with the modules found in the runtime version?

Yep, correct: that pre-list step is needed for runtimes < 3.10, since they lack `sys.stdlib_module_names`.

And yeah, I think we should track expansions on non-Linux distributions in a follow-on -- from a quick look the previous generation strategy didn't cover them at all either :slightly_smiling_face: